system/lib/dlmalloc.c

```diff
@@ -21,6 +21,10 @@
 /* XXX Emscripten Tracing API. This defines away the code if tracing is disabled. */
 #include <emscripten/trace.h>
 
+#ifdef __EMSCRIPTEN_WASM_WORKERS__
+#define USE_LOCKS 1
```

Comment:
Oh, forgot to mention that this change might be a bit scary, since Wasm Workers do not (currently at least) support the full pthread machinery, which includes the pthread_mutex_*() constructs. Though looks like the test suite doesn't fail at least.

It may be that we'll need to enable `USE_SPIN_LOCKS` as well for Wasm Workers, though I'll defer that kind of analysis for until there is a test case that uncovers an issue here.

Reply:
Looks like a duplicate comment, replied above.